modules/apps/content-dashboard/content-dashboard-web/src/main/resources/META-INF/resources/js/components/AuditBarChart.js

```diff
@@ -68,10 +70,26 @@ export default function AuditBarChart({rtl, vocabularies}) {
 			);
 		});
 
-		return {bars, data};
+		const legendCheckboxes = bars.reduce(
+			(acc, {dataKey}) => ({
+				...acc,
+				[dataKey]: true,
+			}),
+			{}
+		);
+
+		const colors = bars.reduce(
+			(acc, {dataKey}, index) => ({
+				...acc,
+				[dataKey]: COLORS[index % COLORS.length],
+			}),
+			{}
+		);
```

Comment:
We could combine these into a single reduce instead of doing it twice, you can do something like


```js
const {legendCheckboxes, colors} = bars.reduce(
	(acc, {dataKey}, index) => ({
		colors: {
			...acc.colors,
			[dataKey]: COLORS[index % COLORS.length],
		},
		legendCheckboxes: {
			...acc. legendCheckboxes,
			[dataKey]: true,
		},
	}),
	{legendCheckboxes: {}, colors: {}}
);
```

Reply:
That's right!